Update the blocks documentation

scruffian opened the pull request:
This should incorporate all the technical details we need to develop blocks in Jetpack.

scruffian committed:
Update the blocks documentation

Co-authored-by: Ben Dwyer <ben@scruffian.com>

roccotripaldi committed:
Extensions Readme

- Making some brief edits
- Filling in holes

With my edits, I'm optimisitically assuming we'll have smooth automation to WordPress.com, even
though it's not currently in place at the time of this writing.

roccotripaldi committed:
Extensions readme

more details about the beta process.

roccotripaldi committed:
Extensions Readme

- More details about releases
- Fix a few typos

roccotripaldi committed:
Extensions Readme

- more details

simison committed:
Typos and backticks

simison committed:
Moar typos and backticks

simison reviewed (dismissed):
This is really nice now!

jeherve commented on extensions/README.md:
```diff
@@ -1,7 +1,64 @@
+- Simply move the extension's slug out of the beta array and into the production array in `extensions/index.json`.
+
+## How do I merge extensions to Jetpack
+- Jetpack is released once a month, so be sure your team is aware of code freeze schedule (PCYsg-eg5-p2).
```
Here I would suggest linking to the public page with our milestones instead of the p2 link, since those include release dates:
https://github.com/Automattic/Jetpack/milestones

jeherve commented on extensions/README.md:
```diff
@@ -1,7 +1,64 @@
+- Once merged, your extension will appear in the next release.
+
+## How do I merge extensions to WordPress.com?
+- Extensions are continuously deployed to WordPress.com, just like other parts of Jetpack.
```
I find this a bit confusing. It makes it sound like it happens automatically, when in fact you need to take action as mentioned below.

jeherve reviewed (dismissed):
Sounds good. I only have a few remarks.

roccotripaldi committed:
Extension Documentation:
Reorganizing headings into a more logical order,
and addressing feedback from @jeherve.

simison reviewed (dismissed):
👍

sirreal commented on extensions/README.md:
````diff
@@ -32,7 +32,72 @@ If your block depends on another block, place them all in extensions folder:
+
+
+```bash
+rsync -a --delete _inc/blocks/ \
````
Minor: Compression is nice (`-z`)

```suggestion
rsync -az --delete _inc/blocks/ \
```

sirreal commented on extensions/README.md:
```diff
@@ -32,7 +32,72 @@ If your block depends on another block, place them all in extensions folder:
+1. Test the WordPress.com diff
+1. Once the code works well in both environments and has been approved by a Jetpack crew member, you can merge your branch!
+
+### BETA Extensions
```
```suggestion
### Beta Extensions
```

sirreal commented on extensions/README.md:
````diff
@@ -32,7 +32,72 @@ If your block depends on another block, place them all in extensions folder:
+You can build extensions from the Jetpack folder to your local sandbox folder and sync the whole sandbox like you always do:
+
+```bash
+yarn build-extensions \
````
```suggestion
yarn clean-extensions
yarn build-extensions \
```

sirreal commented on extensions/README.md:
````diff
@@ -32,7 +32,72 @@ If your block depends on another block, place them all in extensions folder:
+  YOUR_WPCOM_SANDBOX:/BLOCKS_PATH_IN_YOUR_SANDBOX/
+```
+
+Calypso loads Gutenberg from simple sites’ wp-admin in an iframe.
````
```suggestion
To test extensions for a Simple site in Calypso, sandbox the simple site URL (example.wordpress.com). Calypso loads Gutenberg from simple sites’ wp-admin in an iframe.
```

sirreal reviewed (dismissed):
Thanks for this! A few minor notes.

sirreal committed:
BETA → Beta

Co-Authored-By: simison <mikael@ihminen.org>

sirreal committed:
Add -z in rsync

Co-Authored-By: simison <mikael@ihminen.org>

sirreal committed:
clean extensions first

Co-Authored-By: simison <mikael@ihminen.org>

sirreal committed:
Clarification about simple sites

Co-Authored-By: simison <mikael@ihminen.org>

oskosk reviewed (approved):
LGTM
